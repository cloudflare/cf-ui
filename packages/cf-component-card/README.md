# cf-component-card

> Cloudflare Card Component

## Installation

```sh
$ npm install cf-component-card
```

## Usage

<!-- Please add a basic example of how to use this module -->
