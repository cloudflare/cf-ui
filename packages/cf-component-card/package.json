{
  "name": "cf-component-card",
  "description": "CloudFlare Card Component",
  "version": "1.1.1",
  "main": "lib/index.js",
  "module": "es/index.js",
  "jsnext:main": "es/index.js",
  "author": "James Kyle <jkyle@cloudflare.com>",
  "license": "BSD-3-Clause",
  "publishConfig": {
    "registry": "http://registry.npmjs.org/"
  },
  "dependencies": {
    "cf-component-link": "^4.0.1",
    "react": "^0.14.2 || ^15.0.0-0"
  },
  "devDependencies": {
    "assert-equal-jsx": "^1.0.0",
    "cf-component-button": "^3.0.0",
    "react-dom": "^15.0.0-0"
  }
}
