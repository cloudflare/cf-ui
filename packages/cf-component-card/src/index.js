import { applyTheme } from 'cf-style-container';

import CardUnstyled from './Card';
import CardTheme from './CardTheme';
import CardBlockUnstyled from './CardBlock';
import CardBlockTheme from './CardBlockTheme';
import CardContentUnstyled from './CardContent';
import CardContentTheme from './CardContentTheme';
import CardControlUnstyled from './CardControl';
import CardControlTheme from './CardControlTheme';
import CardDrawersUnstyled from './CardDrawers';
import CardDrawersTheme from './CardDrawersTheme.js';
import CardMessagesUnstyled from './CardMessages';
import CardMessagesTheme from './CardMessagesTheme';
import CardSectionUnstyled from './CardSection';
import CardSectionTheme from './CardSectionTheme';
import CardTitleUnstyled from './CardTitle';
import CardTitleTheme from './CardTitleTheme';
import CardFooterUnstyled from './CardFooter';
import CardFooterTheme from './CardFooterTheme';
import CardToolbar from './CardToolbar';
import CardToolbarLink from './CardToolbarLink';
import CardPropTypes from './CardPropTypes';

const Card = applyTheme(CardUnstyled, CardTheme);
const CardBlock = applyTheme(CardBlockUnstyled, CardBlockTheme);
const CardContent = applyTheme(CardContentUnstyled, CardContentTheme);
const CardControl = applyTheme(CardControlUnstyled, CardControlTheme);
const CardDrawers = applyTheme(CardDrawersUnstyled, CardDrawersTheme);
const CardMessages = applyTheme(CardMessagesUnstyled, CardMessagesTheme);
const CardSection = applyTheme(CardSectionUnstyled, CardSectionTheme);
const CardTitle = applyTheme(CardTitleUnstyled, CardTitleTheme);
const CardFooter = applyTheme(CardFooterUnstyled, CardFooterTheme);

export {
  Card,
  CardUnstyled,
  CardTheme,
  CardContent,
  CardContentUnstyled,
  CardContentTheme,
  CardControl,
  CardControlUnstyled,
  CardControlTheme,
  CardDrawers,
  CardDrawersUnstyled,
  CardDrawersTheme,
  CardMessages,
  CardMessagesUnstyled,
  CardMessagesTheme,
  CardPropTypes,
  CardSection,
  CardSectionUnstyled,
  CardSectionTheme,
  CardBlock,
  CardBlockUnstyled,
  CardBlockTheme,
  CardTitle,
  CardTitleUnstyled,
  CardTitleTheme,
  CardFooter,
  CardFooterUnstyled,
  CardFooterTheme,
  CardToolbar,
  CardToolbarLink
};
