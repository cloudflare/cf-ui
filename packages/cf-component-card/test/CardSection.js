import React from 'react';
import renderer from 'react-test-renderer';
import { CardSection } from 'cf-component-card';

test('should render', () => {
  const component = renderer.create(<CardSection>CardSection</CardSection>);
  expect(component.toJSON()).toMatchSnapshot();
});

test('should render with status', () => {
  const component = renderer.create(
    <CardSection status="error">CardSection</CardSection>
  );
  expect(component.toJSON()).toMatchSnapshot();
});
