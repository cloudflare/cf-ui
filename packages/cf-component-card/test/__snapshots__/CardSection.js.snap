// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="cf-card__section cf-card__section--default"
>
  CardSection
</div>
`;

exports[`should render with status 1`] = `
<div
  className="cf-card__section cf-card__section--error"
>
  CardSection
</div>
`;
